### PR TITLE
[1LP][RFR] V2v migration details page

### PR DESCRIPTION
**Pull request by kedark3**
Purpose or Intent
=================

__Adding navigation/Views/Widgets__ for migration plan details page.

**Should be merged after #7343**

### CODE REVIEW

File: widgetastic_manageiq/__init__.py [hunk 1]
```diff
@@ -4356,3 +4356,145 @@ def fill(self, value):
...
+        try:
+            el = self._get_vm_element(vm_name)
+            return self.browser.element(self.ITEM_IS_SUCCESSFUL_LOCATOR,
+             parent=el).is_displayed()
+        except NoSuchElementException:
+            # This means element not present, so plan may not have started
+            return False
+
+
+class MigrationPlanRequestDetailsPaginator(Paginator):
```
Comment (quarckster) [inline widgetastic_manageiq/__init__.py]: **REQUIRED**
It's better to put these pagination widgets into cfme/v2v/migrations.py

File: widgetastic_manageiq/__init__.py [hunk 2]
```diff
@@ -4356,3 +4356,145 @@ def fill(self, value):
...
+        self._click_button('Previous Page')
+
+    def last_page(self):
+        self._click_button('Last Page')
+
+    def first_page(self):
+        self._click_button('First Page')
+
+    def page_info(self):
+        return self.browser.element(self.CUR_PAGE_CTL, parent=self._paginator).text
```
Comment (quarckster) [inline widgetastic_manageiq/__init__.py]: **REQUIRED**
```python
self.browser.text(self.browser.element(self.CUR_PAGE_CTL, parent=self._paginator))
```

File: widgetastic_manageiq/__init__.py [hunk 3]
```diff
@@ -4356,3 +4356,107 @@ def fill(self, value):
...
+        for item in self.browser.elements(self.ITEM_LOCATOR):
+            el = self.browser.element(".//*[@class='list-group-item-heading']", parent=item)
+            if self.browser.text(el) == vm_name:
+                return item
+        raise ItemNotFound("VM: {} not found".format(vm_name))
+
+    def read(self):
+        return self.all_items
+
+    def fill(self, value):
```
Comment (quarckster) [inline widgetastic_manageiq/__init__.py]: **REQUIRED**
In that case do not define `fill()` at all.

**Comment by mshriver** [comment]
WIPTEST, dependent PRs aren't merged

File: widgetastic_manageiq/__init__.py [hunk 4]
```diff
@@ -4356,3 +4356,103 @@ def fill(self, value):
             return False
         self._all_item_elements[value or self.selected_item].click()
         return True
+
+
+class MigrationPlanRequestDetailsList(Widget):
+    """Represents the list of Migration Plan Request Details."""
+
+    ROOT = ParametrizedLocator('.//div[contains(@class,{@list_class|quote})]')
+    ITEM_LOCATOR = ParametrizedLocator('./div[contains(@class,"list-group-item")]')
```
Comment (digitronik) [inline widgetastic_manageiq/__init__.py]: Please, correct xpath. it should be like ".//div[contains(@class,'list-group-item')]"

File: widgetastic_manageiq/__init__.py [hunk 5]
```diff
@@ -4356,3 +4356,103 @@ def fill(self, value):
...
+             parent=el).is_displayed()
+        except NoSuchElementException:
+            # This means element not present, so not in-progress
+            return False
+
+    def is_successful(self, vm_name):
+        """ Checks if check icon is present indicating success."""
+        try:
+            el = self._get_vm_element(vm_name)
+            return self.browser.element(self.ITEM_IS_SUCCESSFUL_LOCATOR,
```
Comment (digitronik) [inline widgetastic_manageiq/__init__.py]: I think you can go with `self.browser.is_displayed(self.ITEM_IS_SUCCESSFUL_LOCATOR, parent=el)`. 
but I will like to go with a widget in the main widget :).

File: widgetastic_manageiq/__init__.py (hunk 4 above)
Comment (kedark3) [reply widgetastic_manageiq/__init__.py]: It does not work with `.//`, hence used, `./`

File: widgetastic_manageiq/__init__.py (hunk 5 above)
Comment (kedark3) [reply widgetastic_manageiq/__init__.py]: Changed this.

File: widgetastic_manageiq/__init__.py (hunk 3 above)
Comment (kedark3) [reply widgetastic_manageiq/__init__.py]: Please review https://github.com/ManageIQ/integration_tests/pull/7343/files for this widget, once that is merged, I will rebase this PR.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by mshriver** [comment]
PRT failure in smoke tests unrelated

File: cfme/v2v/migrations.py [hunk 6]
```diff
@@ -254,6 +298,84 @@ def is_displayed(self):
         return self.title.text == 'Migration Plan Wizard'
 
 
+class MigrationPlanRequestDetailsView(View):
+    migration_request_details_list = MigrationPlanRequestDetailsList("plan-request-details-list")
+    sort_type = SelectorDropdown('id', 'sortTypeMenu')
+    paginator_view = View.include(MigrationPlanRequestDetailsPaginationPane)
```
Comment (mshriver) [inline cfme/v2v/migrations.py]: This should pass `use_parent=True` so that the locators are resolved properly.

Comment (kedark3) [reply cfme/v2v/migrations.py]: @mshriver made this change

**Pull request merged**